### PR TITLE
Add CupertinoScrollbar api docs

### PR DESCRIPTION
**Pull request by Abhishek01039**
Add the sample of the `CupertinoScrollbar` widget

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I signed the [CLA].
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making or feature I am adding, or Hixie said the PR is test-exempt.
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat

### CODE REVIEW

**Comment by Abhishek01039** [comment]
Hello @Piinks 
Can you please review this PR?
Thanks

File: packages/flutter/lib/src/cupertino/scrollbar.dart [hunk 1]
```diff
@@ -38,7 +38,64 @@ const double _kScrollbarCrossAxisMargin = 3.0;
 /// animate from [thickness] and [radius] to [thicknessWhileDragging] and
 /// [radiusWhileDragging], respectively.
 ///
-// TODO(Piinks): Add code sample
+/// {@tool dartpad --template=stateless_widget_scaffold}
+/// This sample shows a [Scrollbar] that executes a fade animation as scrolling occurs.
```
Comment (Piinks) [inline packages/flutter/lib/src/cupertino/scrollbar.dart:42]:
```suggestion
/// This sample shows a [CupertinoScrollbar] that executes a fade animation as scrolling occurs.
```

**Review by Piinks — CHANGES_REQUESTED** [review]
These look like the exact same samples from `Scrollbar`, can we either make them unique? Or define a template so we can re-use without duplicating the code?

**Review by Piinks — COMMENTED** [review]
CupertinoScrollbar has some unique behavior like when the size changes on drag, maybe it would be beneficial to showcase that. :)

File: packages/flutter/lib/src/cupertino/scrollbar.dart (hunk 1 above)
Comment (Abhishek01039) [reply packages/flutter/lib/src/cupertino/scrollbar.dart:42]: Yes I have used the sample code of scrollbar.

**Review by Piinks — COMMENTED** [review]
Hey @Abhishek01039, this does not mention the unique features of the `CupertinoScrollbar`, like `thicknessWhileDragging`?

**Comment by Abhishek01039** [comment]
Ok, let me add this please

File: packages/flutter/lib/src/cupertino/scrollbar.dart [hunk 2]
```diff
@@ -38,7 +38,65 @@ const double _kScrollbarCrossAxisMargin = 3.0;
...
+/// The Scrollbar will fade into view as the user scrolls, and fade out when scrolling stops.
+///
+/// ```dart imports
+/// import 'package:flutter/cupertino.dart';
+/// ```
+///
+/// ```dart
+/// @override
+/// Widget build(BuildContext context) {
+///   return CupertinoScrollbar(
```
Comment (Piinks) [inline packages/flutter/lib/src/cupertino/scrollbar.dart:52]: Let's add those features here too.

File: packages/flutter/lib/src/cupertino/scrollbar.dart [hunk 3]
```diff
@@ -38,7 +38,65 @@ const double _kScrollbarCrossAxisMargin = 3.0;
...
+///     ),
+///   );
+/// }
+/// ```
+/// {@end-tool}
+///
+/// {@tool dartpad --template=stateful_widget_scaffold}
+/// When isAlwaysShown is true, the scrollbar thumb will remain visible without the
+/// fade animation. This requires that a ScrollController is provided to controller,
+/// or that the PrimaryScrollController is available.
```
Comment (Piinks) [inline packages/flutter/lib/src/cupertino/scrollbar.dart:69]: Can you mention the CupertinoScrollbar features you have added in the description?

File: packages/flutter/lib/src/cupertino/scrollbar.dart (hunk 2 above)
Comment (Abhishek01039) [reply packages/flutter/lib/src/cupertino/scrollbar.dart:52]: Yeah good idea
Thanks @Piinks

File: packages/flutter/lib/src/cupertino/scrollbar.dart (hunk 3 above)
Comment (Abhishek01039) [reply packages/flutter/lib/src/cupertino/scrollbar.dart:73]: Can you please give me some hint like what to write in the description?

Comment (Piinks) [reply packages/flutter/lib/src/cupertino/scrollbar.dart:73]: The description should describe the sample. You added `thicknessWhileDragging`, `radiusWhileDragging`, etc.. what do those mean?

**Comment by Abhishek01039** [comment]
Hello @Piinks 
can you please review this PR?
Thanks

File: packages/flutter/lib/src/cupertino/scrollbar.dart [hunk 4]
```diff
@@ -38,7 +38,75 @@ const double _kScrollbarCrossAxisMargin = 3.0;
...
+///         );
+///       },
+///     ),
+///   );
+/// }
+/// ```
+/// {@end-tool}
+///
+/// {@tool dartpad --template=stateful_widget_scaffold}
+/// When isAlwaysShown is true, the scrollbar thumb will remain visible without the
```
Comment (Piinks) [inline packages/flutter/lib/src/cupertino/scrollbar.dart:74]:
```suggestion
/// When `isAlwaysShown` is true, the scrollbar thumb will remain visible without the
```

File: packages/flutter/lib/src/cupertino/scrollbar.dart [hunk 5]
```diff
@@ -38,7 +38,75 @@ const double _kScrollbarCrossAxisMargin = 3.0;
...
+/// fade animation. This requires that a ScrollController is provided to controller,
+/// or that the PrimaryScrollController is available.
```
Comment (Piinks) [inline packages/flutter/lib/src/cupertino/scrollbar.dart:76]:
```suggestion
/// or that the [PrimaryScrollController] is available.
```

**Review by Piinks — COMMENTED** [review]
Hi @Abhishek01039 please check the failing analyzer check

File: packages/flutter/lib/src/cupertino/scrollbar.dart [hunk 6]
```diff
@@ -38,7 +38,75 @@ const double _kScrollbarCrossAxisMargin = 3.0;
...
+/// The thickness of the scrollbar will be 6 when it's ideal and 10 when it's being dragged by the user.
+/// The Radius of the scrollbar thumb's rounded rectangle corners will be 34 when it's ideal and 0
+/// when the scrollbar is being dragged by the user.
```
[the remark this answers is not in the document]
Comment (Abhishek01039) [reply packages/flutter/lib/src/cupertino/scrollbar.dart]: yeah. So let's remove this!

**Review by Piinks — APPROVED** [review]
LGTM 🎉

**Review by Piinks — APPROVED** [review]
LGTM! Thanks for updating!

**Comment by Abhishek01039** [comment]
Thanks @Piinks

**Pull request merged**